### PR TITLE
A sample schema based on a blog

### PR DESCRIPTION
**Pull request by JeffRMoore**
I started developing this schema to teach myself graphql and to understand how one would tackle various data mapping issues when writing a backend.

This one has a many to many relationship, dependent objects, deep nesting of objects, an abstract type and a date value.

I'd guess to meet the criteria of #1 It would need to be augmented with a union type, float, int and ID?  This doesn't have pagination, which should be added.

I think the name of the Topic type is confusing with respect to Category, but couldn't think of anything better.  I do like the name activity as a query operation.

Note this isn't complete and probably doesn't even validate syntax-wise.

### CODE REVIEW

**Comment by koistya** [comment]
For review @sogko

File: SCHEMA.md [hunk 1]
```diff
@@ -1,23 +1,67 @@
...
+  country: String
+}
+
 type User {
-  id: ID!,
-  name: String,
-  email: String,
-  picture(size: Int): Url,
-  friends: [User]!
+  id: String!
```
Comment (koistya) [inline SCHEMA.md]: @JeffRMoore why do you have `String!` instead of `ID!` for identity fields?

File: SCHEMA.md [hunk 2]
```diff
@@ -1,23 +1,67 @@
...
+  name: String
+  location: Location
+  avatar(size: Int): Url,
+  stories: [Story]
```
Comment (koistya) [inline SCHEMA.md]: @JeffRMoore wouldn't it be better to mark array fields as "non null"? E.g. `stories: [Story]!`

**Comment by sogko** [comment]
Hey @JeffRMoore Thanks for the PR, both @koistya and I really appreciate it very much 👍

I've compiled a checklist of things that the schema, that we end up with eventually, should ideally cover.

The idea is to let everyone know what we are missing so that we can add to it in the future.
Also, we hope that it makes it easier for others to contribute to the development of the schema.

Cheers!

----

**Schema checklist**

**Schema**
- [ ] Query
- [ ] Mutation
- [ ] Subscription

**Definitions**
- [ ] GraphQLScalarType
- [ ] GraphQLObjectType
- [ ] GraphQLInterfaceType
- [ ] GraphQLUnionType
- [ ] GraphQLEnumType
- [ ] GraphQLInputObjectType
- [ ] GraphQLList
- [ ] GraphQLNonNull

**Scalars**
- [ ] GraphQLInt
- [ ] GraphQLFloat
- [ ] GraphQLString
- [ ] GraphQLBoolean
- [ ] GraphQLID

**Reference**
http://graphql.org/docs/api-reference-type-system/

File: SCHEMA.md [hunk 3]
```diff
@@ -1,23 +1,67 @@
...
 type User {
-  id: ID!,
-  name: String,
-  email: String,
-  picture(size: Int): Url,
-  friends: [User]!
+  id: String!
+  name: String
+  location: Location
+  avatar(size: Int): Url,
```
Comment (sogko) [inline SCHEMA.md]: I'm assuming that `Url` is a custom defined Scalar type (which is great, it's one of the things we should cover)

If that is so, how about adding the following definition at the top:
```
scalar Url
```

This is the shorthand notation for custom Scalar types.

Cheers!

File: SCHEMA.md [hunk 4]
```diff
@@ -1,23 +1,67 @@
...
+  text: String
+}
+
+type Story : Topic {
+  id: String!
+  author: User
+  dateline: {
+     when: DateTime
+     where: Location
+  }
```
Comment (sogko) [inline SCHEMA.md]: @koistya  @JeffRMoore 

Correct me if I'm wrong, but the shorthand notation does not do nested fields this way?

Probably can re-write this as
```
scalar DateTime // or could be an object type?
type DateLine {
     when: DateTime
     where: Location
}
type Story : Topic {
  id: String!
  author: User
  dateline: DateLine
  title: String
  text: String
  comments: [Comment]
  categories: [Category]
}
```

Cheers!

**Comment by sogko** [comment]
Added note:

I've written a quick cheatsheet on GraphQL shorthand notation over here:
https://wehavefaces.net/graphql-shorthand-notation-cheatsheet-17cd715861b6

Hope that might help with developing the schema.

Cheers!

**Pull request merged**